src/test/java/mate/academy/security/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,79 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.RoleService;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.AdditionalAnswers;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthenticationServiceImplTest {
+    private static final PasswordEncoder PASSWORD_ENCODER = new BCryptPasswordEncoder();
+    private UserService userService;
+    private RoleService roleService;
+    private AuthenticationService authenticationService;
+
+    @BeforeEach
+    void setUp() {
+        userService = Mockito.mock(UserService.class);
+        roleService = Mockito.mock(RoleService.class);
+        authenticationService =
+                new AuthenticationServiceImpl(userService, roleService, PASSWORD_ENCODER);
+    }
```

Comment:
There is no sense to reinstantiate services/mocks before each test if they don't hold any state.
In dao tests such decision was reasonable, but here we have no benefits, only increased tests run time
Should be moved to BeforeAll or intitalized together with variable declaration

Reply:
Thank you! I will keep it in mind.